iothub/device/src/ClientPropertyCollection.cs

```diff
@@ -14,126 +15,104 @@ public class ClientPropertyCollection : PayloadCollection
     {
         private const string VersionName = "$version";
 
-        /// <inheritdoc path="/exception['ArgumentException']" cref="AddInternal(IDictionary{string, object}, string, bool)" />
         /// <summary>
         /// Adds the value to the collection.
         /// </summary>
         /// <remarks>
-        /// If the collection has a key that matches the property name this method will throw an <see cref="ArgumentException"/>.
+        /// If the collection already has a key matching a property name supplied this method will throw an <see cref="ArgumentException"/>.
         /// <para>
         /// When using this as part of the writable property flow to respond to a writable property update you should pass in the value
         /// as an instance of <see cref="PayloadSerializer.CreateWritablePropertyResponse(object, int, long, string)"/>
         /// to ensure the correct formatting is applied when the object is serialized.
         /// </para>
         /// </remarks>
-        /// <exception cref="ArgumentNullException"><paramref name="propertyName"/> is <c>null</c> </exception>
         /// <param name="propertyName">The name of the property to add.</param>
         /// <param name="propertyValue">The value of the property to add.</param>
-        public override void Add(string propertyName, object propertyValue)
-            => Add(null, propertyName, propertyValue);
+        /// <exception cref="ArgumentNullException"><paramref name="propertyName"/> is <c>null</c>. </exception>
+        /// <exception cref="ArgumentException"><paramref name="propertyName"/> already exists in the collection.</exception>
+        public void AddRootProperty(string propertyName, object propertyValue)
```

Comment:
This diff gets a little messy, I'd recommend reviewing the updated file by itself and verifying that the APIs make sense.